sphinxcontrib/confluencebuilder/state.py

```diff
@@ -210,3 +220,32 @@ def upload_id(docname):
         See `registerUploadId` for more information.
         """
         return ConfluenceState.doc2uploadId.get(docname)
+
+    @staticmethod
+    def _format_postfix(postfix, docname, config):
+        """
+        Format a postfix that may have placeholders.
+        All placeholders used must be supported otherwise an error is raised
+        """
+        if postfix and postfix_string_replacement.match(postfix):
```

Comment:
Out of curiosity, does the use of regular expressions here help that much? I assume it was added to help avoid attempts to invoke a format operation (e.g. generating hashes) if no keywords were detected. I would have been happy if we just performed a format request on a non-empty string (even if not keywords were provided), but if the use of `re` helps here, works for me.

That being said, this implementation is calling `match` per title being processed. I would imagine if you are looking for performance here, we could change this to check the postfix once for any possible keywords. And if any are detected, set a state/class flag to be check on each `_format_postfix` call (also, ensuring the flag is cleared in the `reset()` call). Although, I am not worried about it. If you only want to use the implementation as is, that is fine by me.

Lastly, I believe `search` may be a faster alternative to `match` for the small string literals a postfix value could be (based on some casual browsing on the two types).

Reply:
I'm good with keeping this file simple. But ya my reasoning is if we keep adding more replaces placeholders I didn't want to make people using static suffixes builds continually slower. I'll remove all the regex stuff to keep the change straightforward then in the future if more placeholders are added we always could add regex back in and only calculate things that have placeholders present. ie don't calculate hash unless {hash} was found in the postfix. So it would be a dynamically called format